solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -328,6 +329,13 @@ public final void run() {
   }
 
   public final void doRecovery(SolrCore core) throws Exception {
+    if (core.getCoreDescriptor()
```

Comment:
Adding, in our previous queryaggregator implementation we have avoided all these checks by creating proxy core (by overriding few functions in `SolrCore` class). That avoided exposing collection to zookeeper. As exposing collection further creates issue (if app is ZK, or some solr cluster api which returns that collection).

Reply:
These checks are not required. These are just optimisations.